proxy/src/control/discovery.rs

```diff
@@ -209,12 +226,14 @@ where
         };
 
         match update {
-            Update::Insert(addr) => {
-                let service = self.bind.bind(&addr).map_err(|_| ())?;
+            Update::Insert(labeled_addr) => {
+                let service = self.bind.bind(labeled_addr.borrow())
+                    .map(|svc| labeled_addr.label(svc))
+                    .map_err(|_| ())?;
```

Comment:
i'm trying to figure out if it's going to be cleaner to push all of these details behind the bind implementation.

specifically, it would be nice if bind was generic over its address type.  Then, we could, for instance, express requirements on a `Labeled` _trait_ i.e. `B::Service: Labeled`.

i'm going to play with this on a branch so i have a better understanding of the tradeoffs between these approaches.